packages/core/src/suggestion/SuggestionTask.ts

```diff
@@ -0,0 +1,40 @@
+import { performance } from 'perf_hooks';
+
+import { Suggestion } from '../models';
+import { SuggestionInput } from '../types';
+
+interface SuggestionTaskInterface {
+  name: string;
+  elapsedTime: number;
+
+  task?(suggestionInput: SuggestionInput): Promise<Suggestion>;
+}
+
+export class SuggestionTask implements SuggestionTaskInterface {
```

Comment:
Why have an interface here? You can use the class as the type in other places, the interface is moot here, we aren't Java engineers :)

Reply:
updated, it was to not need to implement a default task that didn't do anything.

Reply:
You may want to use an abstract class here instead.

Making `SuggestionTask` an abstract class you can define base level functionality and then declare the task as abstract to have the the plugins define their own.

https://www.typescriptlang.org/docs/handbook/classes.html#abstract-classes

Reply:
okay I will use this.